toucan_connectors/sql_query_helper.py

```diff
@@ -32,12 +32,9 @@ def prepare_limit_query(
         )
         query_check = prepared_query.strip().lower()
         if not query_check.startswith('show') and not query_check.startswith('describe'):
-            if limit and offset:
-                prepared_query = prepared_query.replace(';', '')
-                prepared_query = f'SELECT * FROM ({prepared_query}) LIMIT {limit} OFFSET {offset};'
-            elif limit:
-                prepared_query = prepared_query.replace(';', '')
-                prepared_query = f'SELECT * FROM ({prepared_query}) LIMIT {limit};'
+            if limit:
+                prepared_query = f'SELECT * FROM ({prepared_query.replace(";", "")}) LIMIT {limit}'
+                prepared_query += f' OFFSET {offset};' if offset else ';'
```

Comment:
The "order by" could be very expensive if wwe do it without a limit. So I would recommend doing the `add_default_order_if_needed` only in this `if`.

Reply:
Just to be sure, at the end of our discussion we decided to let this like that right ? 🤔